### ISSUE TITLE
If an agent is hit by an explosion while executing remaining orders in turn based mode, control sometimes returns to the player

### ISSUE THREAD

**Issue by dethstryke01** [OpenApoc/OpenApoc#1419]
This situation reminds me a bit of a thing/bug that existed in the original where you could regain control of units while they were executing remaining orders just by clicking on them (I vote for bringing that back!). 

https://youtu.be/ZhX-Bd77heM?t=1457
[save 36.zip](https://github.com/OpenApoc/OpenApoc/files/15015866/save.36.zip)

OpenApoc 20240410
Extended Weapons Mod 9.01
Falcon Weapons Platform 1.4
Windows 10 22H2
[OpenApoc_settings.zip](https://github.com/OpenApoc/OpenApoc/files/15015884/OpenApoc_settings.zip)

**Comment by FilmBoy84** [comment]
Agreed, being able to manually break execution of remaining orders is really quite important as a feature

Especially as you will see the aliens break the turn flow in the OG during reaction actions - for example being spotted by a player control of the game is sometimes taken away to allow the alien to flee and they can change their original course of this action if they run into another player unit

**Comment by dethstryke01** [comment]
When I'm playing the original I forget to move people who have remaining orders all the time, so it's nice to be able to regain control of them while they're moving and even check a second time to see if I've forgotten anyone else. It was never clear to me whether this was intended or a bug in the game.

**Comment by FilmBoy84** [comment]
It's definitely a feature - and one offered to "all sides" when things are working as intended, hence my above example
I think the intent was to make TB more "fluid" and dynamic than the simple reaction fire stuff we saw in UFO/TFTD
There's a lot more freedom with what TUs can be spent on both with remaining orders and reactions

**Comment by dethstryke01** [comment]
Cool. If it's not a bug, I think it should be added to OpenApoc. I'll make a video showing a few examples of how it works and post a feature request.
